### PR TITLE
CHK-6821: Fix Place Order Error When Agreements Are Not Accepted

### PR DESCRIPTION
**Pull request by henry-bold**
Fixes [CHK-6821](https://boldapps.atlassian.net/browse/CHK-6821)

[CHK-6821]: https://boldapps.atlassian.net/browse/CHK-6821?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by JosephLeedy — DISMISSED** [review]

File: view/frontend/web/js/view/payment/method-renderer/bold-spi.js [hunk 1]
```diff
@@ -100,6 +117,13 @@ define([
             const isFastlaneAvailable = fastlane.isAvailable();
             this.isSpiLoading(false);
 
+            if (localStorage.getItem('checkoutAcceptedAgreement') === '1') {
+                document.querySelectorAll('input[data-gdpr-checkbox-code="privacy_checkbox"],' +
```
[the remark this answers is not in the document]
Comment (henry-bold) [reply view/frontend/web/js/view/payment/method-renderer/bold-spi.js:121]: both cases. the GDPR is when you use Amasty, and the input[data-gdpr-checkbox-code="privacy_checkbox"] is when you don't use it. So, it is to work in both cases

File: view/frontend/web/js/view/payment/method-renderer/bold-spi.js [hunk 2]
```diff
@@ -28,8 +30,23 @@ define([
...
+
+    const validateAgreements = () => {
+        if (!additionalValidators.validate()) {
+            messageList.addErrorMessage({
+                message: $t('Please agree to all the terms and conditions before placing the order.')
+            });
+            localStorage.removeItem('checkoutAcceptedAgreement');
+            return false;
+        }
+        localStorage.setItem('checkoutAcceptedAgreement', '1');
```
[the remark this answers is not in the document]
Comment (davemacaulay) [reply view/frontend/web/js/view/payment/method-renderer/bold-spi.js:46]: Ah okay got it, that makes sense and is certainly annoying.

I think we should probably clear this at the end of Checkout on the final steps, just to ensure any new Checkout requires them to confirm again.

[the remark this answers is not in the document]
Comment (JosephLeedy) [reply view/frontend/web/js/view/payment/method-renderer/bold-spi.js:46]: Good idea, @davemacaulay!

[the remark this answers is not in the document]
Comment (henry-bold) [reply view/frontend/web/js/view/payment/method-renderer/bold-spi.js:46]: The problem is.. when everything goes fine it (the backend) goes to the "successful page", when there is an issue it sends the customer back to the checkout page. Right now the only layout we touch is checkout_index_index.xml.  I would need to create one more, is that ok?

**Review by bold-eric-d — APPROVED** [review]

File: view/frontend/web/js/view/payment/method-renderer/bold-spi.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (henry-bold) [reply view/frontend/web/js/view/payment/method-renderer/bold-spi.js]: @davemacaulay  Updated the code. Now it will delete it.

**Review by davemacaulay — DISMISSED** [review]

**Review by JosephLeedy — APPROVED** [review]

**Review by nicolenorman — APPROVED** [review]

**Review by TanyaP-sdet — APPROVED** [review]

**Review by bold-eric-d — APPROVED** [review]

**Review by JosephLeedy — APPROVED** [review]

**Review by davemacaulay — APPROVED** [review]

**Pull request merged**